disnake/ui/modal.py

```diff
@@ -0,0 +1,272 @@
+
+        Parameters
+        ----------
+        label: :class:`str`
+            The label of the input text.
+        custom_id: :class:`str`
+            The ID of the input text that gets received during an interaction.
+        style: :class:`.InputTextStyle`
+            The style of the input text.
+        placeholder: Optional[:class:`str`]
+            The placeholder text that is shown if nothing is entered.
+        value: Optional[:class:`str`]
+            The pre-filled value of the input text.
+        required: :class:`bool`
+            Whether the input text is required. Defaults to ``True``.
+        min_length: :class:`int`
+            The minimum length of the input text. Defaults to ``0``.
+        max_length: Optional[:class:`int`]
+            The maximum length of the input text.
+        """
+        self.components.append(
+            ActionRow(
+                InputText(
+                    label=label,
+                    custom_id=custom_id,
+                    style=style,
+                    placeholder=placeholder,
+                    value=value,
+                    required=required,
+                    min_length=min_length,
+                    max_length=max_length,
+                )
+            )
+        )
+
+    async def callback(self, interaction: ModalInteraction) -> None:
+        """|coro|
+
+        The callback associated with this modal.
+
+        This can be overriden by subclasses.
+
+        Parameters
+        ----------
+        interaction: :class:`.ModalInteraction`
+            The interaction that triggered this modal.
+        """
+        pass
+
+    async def on_error(self, error: Exception, interaction: ModalInteraction) -> None:
+        """|coro|
+
+        A callback that is called when an error occurs.
+
+        The default implementation prints the traceback to stderr.
+
+        Parameters
+        ----------
+        error: :class:`Exception`
+            The exception that was raised.
+        interaction: :class:`.ModalInteraction`
+            The interaction that triggered this modal.
+        """
+        traceback.print_exception(error.__class__, error, error.__traceback__, file=sys.stderr)
+
+    async def on_timeout(self) -> None:
+        """|coro|
+
+        A callback that is called when the modal is removed from the cache
+        without an interaction being made.
+        """
+        pass
+
+    def to_components(self) -> ModalPayload:
+        payload: ModalPayload = {
+            "title": self.title,
+            "custom_id": self.custom_id,
+            "components": [component.to_component_dict() for component in self.components],
+        }
+
+        return payload
+
+    async def _scheduled_task(self, interaction: ModalInteraction) -> None:
+        try:
+            await self.callback(interaction)
+        except Exception as e:
+            await self.on_error(e, interaction)
+        else:
```

Comment:
There are some (edge??) cases when the error is more the fault of the user's input than the developer's code.

As an example, let's use the create tag implementation:

if I have two inputs, one with style short and with a max of 50 characteres,
and another one with style long but without a maximum characters (4k by default).

if the user sends a 1024+ string on the second input and then I use that and put it in an embed, it would raise an error, but then if I send a -1024 string, the modal would be sent successfully.

(I hope I explained correctly)

Reply:
I see, that makes sense
For some reason I incorrectly assumed that the modal would always close when you submit it, but it only closes when a response is received from the bot, so that `else` is correct